packages/dom/src/dom.js

```diff
@@ -143,20 +143,24 @@ function isEdge( container, isReverse, onlyVertical ) {
 		return true;
 	}
 
+	// In the case of RTL scripts, the horizontal edge is at the opposite side.
+	const { direction } = computedStyle;
+	const isReverseDir = direction === 'rtl' ? ( ! isReverse ) : isReverse;
```

Comment:
Isn't `isReverse` already prepared for RTL at this point, or at least in how it's called from `WritingFlow`? In which case, wouldn't reversing it once more here cause it to revert back to being wrongly treated as LTR?

Reply:
No, I think this is different? In RTL the text is aligned right, so instead of setting test ranges at the left, we need to set it to the right.

Reply:
> No, I think this is different? In RTL the text is aligned right, so instead of setting test ranges at the left, we need to set it to the right.

I agree it needs to be reversed. But if I'm following correctly, it is already reversed by this point when called from `WritingFlow`. So reversing it again would be wrong?

Logic flow:

1. https://github.com/WordPress/gutenberg/blob/e2b12c862c72aaba29d179b7856bdca32cef0858/packages/block-editor/src/components/writing-flow/index.js#L293
2. https://github.com/WordPress/gutenberg/blob/e2b12c862c72aaba29d179b7856bdca32cef0858/packages/block-editor/src/components/writing-flow/index.js#L324
3. https://github.com/WordPress/gutenberg/blob/e2b12c862c72aaba29d179b7856bdca32cef0858/packages/dom/src/dom.js#L177-L178
4. https://github.com/WordPress/gutenberg/blob/e2b12c862c72aaba29d179b7856bdca32cef0858/packages/dom/src/dom.js#L148

Reply:
They are two different things.

The second parameter of `isHorizontalEdge` is used to signal which *side* of the field to check (before first character or after the last).

In RTL, left and right are reversed at the edges:

* When pressing arrow RIGHT at the start, selection should move to the previous block.
* When pressing arrow LEFT at the end, selection should move to the next block.

INSIDE `isHorizontalEdge`, it does something different.

If you want to see if the selection is at the start OR end, it should check the right side for RTL, but left side for LTR.

I hope this makes sense. Maybe I should inline comment. :)